django_rich_logging/logging.py

```diff
@@ -34,34 +49,71 @@ def __init__(self, *args, **kwargs):
             self.console = Console()
 
         if self.live is None:
-            self.uri_table = Table()
+            self.table = Table()
+            self._add_column_headers()
+
+            self.live = Live(self.table, auto_refresh=False)
+
+    def _add_column_headers(self):
+        """
+        Add column headers for each column.
+        """
+
+        for column in self.columns:
+            self.table.add_column(column["header"])
+
+    def _add_row(self, request_record: RequestRecord) -> None:
+        """
+        Add a row to the live updating table that includes rendered markup for each column.
+        """
 
-            # TODO: Add columns based on config
-            self.uri_table.add_column("Method")
-            self.uri_table.add_column("URI")
-            self.uri_table.add_column("Status")
-            self.uri_table.add_column("Size")
-            self.uri_table.add_column("Time")
+        row_columns = []
 
-            self.live = Live(self.uri_table, auto_refresh=False)
+        for column in self.columns:
+            markup = self._get_markup(request_record, column)
+            text = Text.from_markup(markup, style=request_record.text_style)
+
+            row_columns.append(text)
+
+        self.table.add_row(*row_columns)
+
+    def _get_markup(self, request_record: RequestRecord, column: Dict) -> str:
+        """
+        Get markup based on the column's `format` and the `RequestRecord` object.
+        """
+
+        format = column.get("format", "")
+        style = column.get("style", "%")
+        date_format = column.get("datefmt")
+
+        if style == "%":
+            return format % request_record.get_dict(date_format=date_format)
+        elif style == "{":
+            return format.format(**request_record.get_dict(date_format=date_format))
+        elif style == "$":
+            template = Template(format)
+
+            return template.substitute(
+                **request_record.get_dict(date_format=date_format)
+            )
```

Comment:
I know this PR is still in the works, so a minor suggestion here - you can store the return value of `request_record.get_dict(date_format=date_format)` in a variable instead of having duplicate calls?

Reply:
Oops, I didn't see this before I merged. I think you are right, that probably would be cleaner! I'll make this tweak. 👍

Reply:
Done in 2d2682e.

Reply:
awesome!!